CI tests are failing with "ResourceWarning: unclosed scandir iterator"

`test_vcfzarr_to_zarr` is failing with fairly high frequency on the main build. E.g. https://github.com/pystatgen/sgkit/runs/2981156269?check_suite_focus=true

```
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <posix.ScandirIterator object at 0x7f869c1281f0>
E               
E               Traceback (most recent call last):
E                 File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/fsspec/spec.py", line 394, in walk
E                   return path, {}, {}
E               ResourceWarning: unclosed scandir iterator <posix.ScandirIterator object at 0x7f869c1281f0>
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -7,4 +7,4 @@ scipy
 typing-extensions
 numba
 zarr
-fsspec
+fsspec != 2021.6.*

--- a/setup.cfg
+++ b/setup.cfg
@@ -35,7 +35,7 @@ install_requires =
     zarr
     numba
     typing-extensions
-    fsspec
+    fsspec != 2021.6.*
     setuptools >= 41.2  # For pkg_resources
 setup_requires =
     setuptools >= 41.2

